src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/custom_nonce.py

```diff
@@ -0,0 +1,63 @@
+
+
+if get_api_version(ResourceType.MGMT_STORAGE_STORAGE_ACCOUNTS) in ['2016-12-01']:
+
+    def create_storage_account(resource_group_name, account_name, sku, location,
+                               kind=get_versioned_models(ResourceType.MGMT_STORAGE_STORAGE_ACCOUNTS, model='Kind').storage.value, tags=None, custom_domain=None,
+                               encryption=None, access_tier=None):
+        StorageAccountCreateParameters = get_versioned_models(ResourceType.MGMT_STORAGE_STORAGE_ACCOUNTS, model='StorageAccountCreateParameters')
+        Kind = get_versioned_models(ResourceType.MGMT_STORAGE_STORAGE_ACCOUNTS, model='Kind')
+        Sku = get_versioned_models(ResourceType.MGMT_STORAGE_STORAGE_ACCOUNTS, model='Sku')
+        CustomDomain = get_versioned_models(ResourceType.MGMT_STORAGE_STORAGE_ACCOUNTS, model='CustomDomain')
+        AccessTier = get_versioned_models(ResourceType.MGMT_STORAGE_STORAGE_ACCOUNTS, model='AccessTier')
+        scf = storage_client_factory()
+        params = StorageAccountCreateParameters(
+            sku=Sku(sku),
+            kind=Kind(kind),
+            location=location,
+            tags=tags,
+            custom_domain=CustomDomain(custom_domain, None) if custom_domain else None,
+            encryption=encryption,
+            access_tier=AccessTier(access_tier) if access_tier else None)
+        return scf.storage_accounts.create(resource_group_name, account_name, params)
+
+    def update_storage_account(instance, sku=None, tags=None, custom_domain=None,
+                               use_subdomain=None, encryption=None, access_tier=None):
+        StorageAccountUpdateParameters = get_versioned_models(ResourceType.MGMT_STORAGE_STORAGE_ACCOUNTS, model='StorageAccountUpdateParameters')
+        Sku = get_versioned_models(ResourceType.MGMT_STORAGE_STORAGE_ACCOUNTS, model='Sku')
+        CustomDomain = get_versioned_models(ResourceType.MGMT_STORAGE_STORAGE_ACCOUNTS, model='CustomDomain')
+        AccessTier = get_versioned_models(ResourceType.MGMT_STORAGE_STORAGE_ACCOUNTS, model='AccessTier')
+        domain = instance.custom_domain
+        if custom_domain is not None:
+            domain = CustomDomain(custom_domain)
+            if use_subdomain is not None:
+                domain.name = use_subdomain == 'true'
+
+        params = StorageAccountUpdateParameters(
+            sku=Sku(sku) if sku is not None else instance.sku,
+            tags=tags if tags is not None else instance.tags,
+            custom_domain=domain,
+            encryption=encryption if encryption is not None else instance.encryption,
+            access_tier=AccessTier(access_tier) if access_tier is not None else instance.access_tier
+        )
+        return params
+
+elif get_api_version(ResourceType.MGMT_STORAGE_STORAGE_ACCOUNTS) in ['2015-06-15']:
+
+    def create_storage_account(resource_group_name, account_name, location, account_type, tags=None):
```

Comment:
I would suggest this will be a tricket wicket to solve. I'd be fine with merging the PR, creating issues for the rough spots, and then, once those issues are fixed, moving forward with the wholesale implementation.